src/main/java/games/strategy/triplea/ui/EditPanel.java

```diff
@@ -272,17 +271,8 @@ public void actionPerformed(final ActionEvent event) {
           cancelEditAction.actionPerformed(null);
           return;
         }
-        final Set<TechAdvance> advance = new HashSet<>();
-        // FIXME: This try-catch block appears to be guarding against a ClassCastException.
-        // It should be removed and the underlying problem fixed.
-        try {
-          for (final Object selection : techList.getSelectedValuesList()) {
-            advance.add((TechAdvance) selection);
-          }
-        } catch (final Exception e) {
-          ClientLogger.logQuietly("Failed to copy all selected tech advances", e);
-        }
-        final String result = EditPanel.this.frame.getEditDelegate().addTechAdvance(player, advance);
+        final Set<TechAdvance> techAdvances = new HashSet<>(techList.getSelectedValuesList());
```

Comment:
This line is simply removing duplicates.  It doesn't appear that duplicates are possible in `techList`, but I didn't dig too deeply into confirming this assertion, so I just left this line intact.